### DISCUSSION TITLE
Error on MongoDB

### DISCUSSION THREAD

**Discussion by NOMAD11TH** [Arkensor/EnfusionPersistenceFramework#34] [Q&A]
Greetings,

I am running into the following issue when starting my server with your mission utilizing MongoDB, it appears its unable to write to the database and I am not sure what I am doing wrong. Any help would be appreciated.

I am receiving the following errors on repeat on the logs on both workbench and live testing on a dedicated server with both GPortal and GTXGaming:
22:00:06.899 BACKEND   (E): Curl error=URL using bad/illegal format or missing URL
22:00:06.912 BACKEND   (E): Http Error (BAD_REQUEST)
22:00:06.926 BACKEND   (E): [RestApi] ID:[19] Error - Cannot send request

EDF_MongoDBConnectioninfo Settings:
Database Name: 11THABRNEDatabase
Proxy Host: ac-nobyiox-shard-00-01.uv35ozb.mongodb.net
Proxy Port: 27017
Secure Connection {Unchecked}
Parameters (0)

**Comment by Arkensor** [comment]
Hmmmm can you try it out with a local mongodb installation? I'll check what this is about as soon as I have time thx

**Reply by Braanflakes** [reply]
I was just trying a remote mongodb installation as well and am running into similar errors. I set up the mongodb instance to be as simple as possible with no authentication enabled and I can confirm it's making a valid connection to the database, however I get the following error 5 times in a row in rapid succession as it tries to send/receive data (I assume). The "bad request" response tells me the request is malformed in some way but I really don't know how, unless I'm supposed to be providing any headers/params to the request, I'm just not sure what those would be.

BACKEND   (E): Curl error=Failure when receiving data from the peer
BACKEND   (E): Http Error (BAD_REQUEST)

Additionally, I tried providing the -ConnectionString CLI option in various formats with no luck either. If you have any example connection strings that are working not using localhost, I'd love to see them so I can get an idea of what I'm supposed to be using here.

Lastly, I did try running a MongoDB server locally which let me get past the initial errors I was having but then I get a null reference error for dbContext inside of the AfterWorldPostProcess() function in EPF_PersistenceManager.

Thank you!

**Reply by Arkensor** [reply]
Regarding the first I would need more info, please. Did you try to connect to a cloud mongo db instance directly? You need to use the proxy app either way. Maybe that is why you got errors?

For the nullptr exception, I need the full stack trace of the lo,g please.
